WIP: Self deposit rebase  (Dont merge yet)

emudojo opened the pull request.

emudojo committed:
doi populator

(cherry picked from commit e79736e8ab6a7a62c2a8569176f1cb68fcb180bd)

emudojo committed:
doi populator

(cherry picked from commit 12ce366ff314c6a3c158d2f8181ce61d99a79a04)

emudojo committed:
Added mappings to custom fields
Added hooks to webform to enable custom fields and mimeTypes

(cherry picked from commit 760c93167e5addb42ef0df7e746c53c0749a8bc1)

emudojo committed:
- Added install file
- removed un-used fields in configuration
- Updated eleible types hook

emudojo committed:
- small update to get rid of warnings and ignore un-mapped fields

emudojo committed:
- added exception to ingest page
- added new module to set embargo options for ingested objects

emudojo committed:
saved embargo, setting from the field

emudojo committed:
- Added U1-U2 fields WIP

emudojo committed:
- fixed saving not working properly on single value fields
- Added islandora_webform parameter to distinguish a ingest when adding U1/2 fields
- Code cleanup, removed most warnings

emudojo committed:
serveral fixes

emudojo committed:
fixed merge conflict

emudojo committed:
- quick fix to allow individual values to be added to the preview form.

emudojo committed:
Updated req modules to match

sarah-lippincott committed:
Create help_with_icg_webform_selfsubmit.md

dwk2 commented:
Hi @dmoses, can you take a look at this too, please?

dwk2 committed:
Merge pull request #46 from sarah-lippincott/patch-1

Create help_with_icg_webform_selfsubmit.md

dwk2 commented:
Hi @cbarnesphd, can you look at this, too?

cbarnesphd commented on docs/help_with_icg_webform_selfsubmit.md:
```diff
@@ -0,0 +1,74 @@
+### Islandora Settings
+Configure how the form interacts with your Islandora collections at webform/islandora. **You must configure these settings before you will be able to map your form components to MODS metadata fields.**
+
+* Select whether to allow the form to show up as a link on certain object pages. The typical self-deposit workflow will likely **not** use this option. 
```
This could use some further explanation. If I want to be able to add a link to only certain collections (i.e. our IR collection), would I want to check 'Enable'? I believe so, because David's IW documentation says that I need to manually tag pages on which I want the link to appear.

noahwsmith replied:
TODO: clarify how to get the URL of the webform and add it as a link elsewhere on the site.

cbarnesphd commented on docs/help_with_icg_webform_selfsubmit.md:
```diff
@@ -0,0 +1,74 @@
+* **Content Model Filter:** Select a content model for the objects this form will ingest. Each form can only support one content model, meaning a distinct form will be needed for each media type.
+* **Collection Filter:** Select the collection where submissions will be deposited. Each form can only support deposit to one collection.
+* **Islandora Embargo Settings:** Set global embargo settings for objects submitted via this form or select "Add embargo field to the form" to allow users to set their own embargo. 
+* **Enable Ingest:** Select this option to allow users to upload files. Select the appropriate content model from the Ingest Destination drop-down. Set the relationship of your new object to the collection (typically isMemberOf), and define the appropriate namespace.
```
The fact that "current object" refers to the collection selected in the Collection Filter could be clarified here.

patdunlavey replied:
Chris, I don't see where the text "current object" appears in this file. Can you clarify what you are referring to?

> The fact that "current object" refers to the collection selected in the Collection Filter could be clarified here.

noahwsmith replied:
TODO - change helptext / labels on https://lasir-demo.born-digital.com/node/14/webform/islandora ("Relationship to current object")

emudojo committed:
#462312 Fixed a problem when saving an embargo date w/o a date

emudojo committed:
Merge remote-tracking branch 'origin/self-deposit-rebase' into self-deposit-rebase

emudojo committed:
fixed embargo date when empty, added sid column to select the correct configuration

emudojo committed:
Misc fixes to embargo, removed warnings and WSOD

emudojo committed:
Moved imports to the general scope of the function to avoid errors when using submit functions

sarah-lippincott committed:
Update help_with_icg_webform_selfsubmit.md

noahwsmith committed:
Merge pull request #48 from sarah-lippincott/patch-3

Merging into outstanding PR #45

emudojo committed:
- remove warnings when submitting objects

emudojo committed:
Merge remote-tracking branch 'origin/self-deposit-rebase' into self-deposit-rebase

emudojo committed:
removed pager from query as it is not working

patdunlavey committed:
pat / self-deposit-rebase --- rewrite of iw_get_collections_or_content_models
The collections list did not account for namespace restrictions. This rewrite changes the collections search to use
solr, and adds namespace restriction to the query.

dwk2 committed:
Merge pull request #49 from patdunlavey/self-deposit-rebase

pat / self-deposit-rebase --- rewrite of iw_get_collections_or_content_models()

emudojo committed:
several fixes to embargo an general logic

emudojo committed:
Merge branch 'self-deposit-rebase' of github.com:Islandora-Collaboration-Group/islandora_webform into self-deposit-rebase

emudojo committed:
removed array definition

noahwsmith reviewed (approved):
This is ready to go based on our testing.
